tools/pubsys/src/aws/validate_ami/results.rs

```diff
@@ -0,0 +1,714 @@
+            results.extend(
+                region_results
+                    .iter()
+                    .filter(|result| requested_status.contains(&result.status))
+                    .collect::<HashSet<&AmiValidationResult>>(),
+            )
+        }
+        results
+    }
+
+    fn get_results_summary(&self) -> HashMap<Region, AmiValidationRegionSummary> {
+        self.results
+            .iter()
+            .map(|(region, region_result)| {
+                region_result
+                    .as_ref()
+                    .map(|region_validation| {
+                        (
+                            region.clone(),
+                            AmiValidationRegionSummary::from(region_validation),
+                        )
+                    })
+                    .unwrap_or((
+                        region.clone(),
+                        AmiValidationRegionSummary::no_valid_results(),
+                    ))
+            })
+            .collect()
+    }
+
+    pub(crate) fn get_json_summary(&self) -> serde_json::Value {
+        serde_json::json!(self
+            .get_results_summary()
+            .into_iter()
+            .map(|(region, results)| (region.to_string(), results))
+            .collect::<HashMap<String, AmiValidationRegionSummary>>())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::{AmiValidationResult, AmiValidationResultStatus, AmiValidationResults};
+    use crate::aws::validate_ami::ami::ImageDef;
+    use aws_sdk_ssm::Region;
+    use std::collections::{HashMap, HashSet};
+
+    // These tests assert that the `get_results_for_status` function returns the correct values.
+
+    // Tests empty AmiValidationResults
+    #[test]
+    fn get_results_for_status_empty() {
+        let results = AmiValidationResults::new(HashMap::from([
+            (Region::new("us-west-2"), Ok(HashSet::from([]))),
+            (Region::new("us-east-1"), Ok(HashSet::from([]))),
+        ]));
+        let results_filtered = results.get_results_for_status(&vec![
+            AmiValidationResultStatus::Correct,
+            AmiValidationResultStatus::Incorrect,
+            AmiValidationResultStatus::Missing,
+        ]);
+
+        assert_eq!(results_filtered, HashSet::new());
+    }
+
+    // Tests the `Correct` status
+    #[test]
+    fn get_results_for_status_correct() {
+        let results = AmiValidationResults::new(HashMap::from([
+            (
+                Region::new("us-west-2"),
+                Ok(HashSet::from([
+                    AmiValidationResult::new(
+                        "test3-image-id".to_string(),
+                        ImageDef::expected("test3-image-id".to_string()),
+                        Some(ImageDef {
+                            image_id: "test3-image-id".to_string(),
+                            public: true,
+                            ena_support: false,
+                            sriov_net_support: "simple".to_string(),
```

Comment:
Do we actually want to be including these fields in our configuration files and comparisons? @cbgbt @rpkelly @yeazelm Should we be simply validating that an AMI_ID is public, or should we be including these extra validations (I don't know what ena_support or sriov_net_support are, and I suspect we just get whatever is default for these)

Reply:
This was @cbgbt 's suggestion, because we know for all Bottlerocket AMIs that ena_support should be true and sriov_net_support should be simple.

Reply:
These are unlikely to be wrong, but a big deal if they are. So its worth ensuring the state of the AMI is not only that it exists, but has the right fields set. I'm a fan of including them if it doesn't affect complexity too much.

Reply:
Ok, sounds good he knows more than me.